smac/runhistory/runhistory.py

```diff
@@ -103,19 +103,27 @@ def add(self, config, cost, time,
         v = RunValue(cost, time, status, additional_info)
 
         # Each runkey is supposed to be used only once. Repeated tries to add
-        # the same runkey will be ignored silently.
-        if self.data.get(k) is None or self.data.get(k).status==StatusType.CAPPED:
-            self.data[k] = v
+        # the same runkey will be ignored silently if not capped.
+        if self.data.get(k) is None:
+            self._add(k, v, status, external_data)
+        elif status != StatusType.CAPPED and self.data[k].status == StatusType.CAPPED:
```

Comment:
If the run is still capped but with a higher cutoff, shouldn't we store it? Can that happen?